File: algorithms/1848.minimum-distance-to-the-target-element/solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @param {Integer} start
# @return {Integer}
def get_min_distance(nums, target, start)
  (0...nums.size)
    .select {|i| nums[i] == target }
    .map {|n| (n - start).abs }
    .min
end

File: algorithms/1848.minimum-distance-to-the-target-element/solution.rs
impl Solution {
    pub fn get_min_distance(nums: Vec<i32>, target: i32, start: i32) -> i32 {
        (0..nums.len())
            .collect::<Vec<_>>()
            .into_iter()
            .filter(|&i| nums[i] == target)
            .into_iter()
            .map(|n| (n as i32 - start).abs())
            .into_iter()
            .min()
            .unwrap() as i32
    }
}
